src/MessageHandler/LostMessagesConsumer.php

```diff
@@ -44,33 +44,50 @@ public function setMessageTtl(int $messageTtl): void
     /**
      * Requeue message with date superior to message ttl
      * Then max retry reached, message is put in dead letter list
+     *
+     * @param int|null $secondTime
      */
-    public function requeueOldMessages()
+    public function requeueOldMessages(?int $secondTime = null): void
     {
-        array_map(
-            [$this, 'requeueOldMessageFromList'],
-            iterator_to_array($this->queue->getWorkingLists($this->redisClient))
-        );
+        array_map([$this, 'requeueOldMessageFromList'], iterator_to_array($this->queue->getWorkingLists($this->redisClient)), [$secondTime]);
     }
 
-    protected function requeueOldMessageFromList(string $list)
+    /**
+     * Requeue old messages from $list, and remove them if
+     * $messageTTL or time is exceed.
+     *
+     * @param string   $list
+     * @param int|null $secondTime
+     */
+    protected function requeueOldMessageFromList(string $list, ?int $secondTime = null)
     {
-        for ($i = 0, $n = $this->redisClient->llen($list); $i < $n; $i++) {
-            $serializedMessage = $this->redisClient->rpoplpush($list, $list);
+        while ($serializedMessage = $this->redisClient->rpoplpush($list, $list))
+        {
+            // Break the loop, if message is empty.
+            if (empty($serializedMessage)){
+                break;
+            }
+
+            // If maximum times is defined and time is exceed.
+            if (($secondTime !== null && time() >= (time() + $secondTime))) {
```

Comment:
You should add or make a test for this "if" :trollface: 
Please read it again ;-)

Reply:
And maybe better at the end of the loop